neuro_config_client/factories.py

```diff
@@ -124,57 +118,15 @@ def _create_vcd_cloud_provider_options(
     @staticmethod
     def create_node_pool_options(payload: dict[str, Any]) -> NodePoolOptions:
         return NodePoolOptions(
-            id=payload["id"],
             machine_type=payload["machine_type"],
             cpu=payload["cpu"],
-            available_cpu=payload["available_cpu"],
+            available_cpu=payload.get("available_cpu"),
             memory=payload["memory"],
-            available_memory=payload["available_memory"],
-            gpu=payload.get("gpu"),
-            gpu_model=payload.get("gpu_model"),
-        )
-
-    @classmethod
-    def create_storage_options(
-        cls, type: CloudProviderType, payload: dict[str, Any]
-    ) -> StorageOptions:
-        if type == CloudProviderType.AWS:
-            return cls.create_aws_storage_options(payload)
-        elif type == CloudProviderType.GCP:
-            return cls.create_google_storage_options(payload)
-        elif type == CloudProviderType.AZURE:
-            return cls.create_azure_storage_options(payload)
-        else:
-            raise ValueError(
-                f"Storage options are not supported for {type.value!r} cloud provider"
-            )
-
-    @staticmethod
-    def create_aws_storage_options(payload: dict[str, Any]) -> AWSStorageOptions:
-        return AWSStorageOptions(
-            id=payload["id"],
-            performance_mode=EFSPerformanceMode(payload["performance_mode"]),
-            throughput_mode=EFSThroughputMode(payload["throughput_mode"]),
-            provisioned_throughput_mibps=payload.get("provisioned_throughput_mibps"),
-        )
-
-    @staticmethod
-    def create_google_storage_options(payload: dict[str, Any]) -> GoogleStorageOptions:
-        return GoogleStorageOptions(
-            id=payload["id"],
-            tier=GoogleFilestoreTier(payload["tier"]),
-            min_capacity=payload["min_capacity"],
-            max_capacity=payload["max_capacity"],
-        )
-
-    @staticmethod
-    def create_azure_storage_options(payload: dict[str, Any]) -> AzureStorageOptions:
-        return AzureStorageOptions(
-            id=payload["id"],
-            tier=AzureStorageTier(payload["tier"]),
-            replication_type=AzureReplicationType(payload["replication_type"]),
-            min_file_share_size=payload["min_file_share_size"],
-            max_file_share_size=payload["max_file_share_size"],
+            available_memory=payload.get("available_memory"),
+            nvidia_gpu=payload.get("nvidia_gpu") or payload.get("gpu"),
+            nvidia_gpu_model=(
+                payload.get("nvidia_gpu_model") or payload.get("gpu_model")
+            ),
```

Comment:
@zubenkoivan Can you confirm please, we shouldn't include AMD and intel here because cloud providers doesn't have them, right?

Reply:
yes, correct